### DISCUSSION TITLE
[Question] Can Tauri bundle existing websites into desktop apps?

### DISCUSSION THREAD

**Discussion by RpxdYTX** [tauri-apps/tauri#4734] [Q&A] [answered]
Not that relevant, and maybe i should've used another way of asking questions rather than gh's issues, but given some context...

Well, an application on npm, called [nativefier](https://github.com/nativefier/nativefier), is able to make websites into desktop applications using Electron.
...yeah that's the context.

Since Tauri kinda does something similar to Electron, but using different libraries to get better performance and binary sizes, can Tauri bundle existing websites that way? (Note: not a js guy, didn't read nor understood the whole nativefier's source code, even though it seems to be pretty straightforward, so idk whether or not that part of bundling is a special magic done on nativefier's implementation itself but yeah)

Just curiosity, not something urgent whatsoever.

**Comment by ColonelThirtyTwo** [comment, marked as answer]
You'd probably HTTP request intercepting for this to work. I don't think Tauri implements that yet.

**Reply by RpxdYTX** [reply]
I was thinking that out of random ideas, but thanks.

**Comment by FabianLars** [comment]
Generally yes, but there was no work done to streamline use-cases like this (especially since we completely consider browser(-like) use-cases out of scope for now), so expect rough edges here and there (or missing features, i'm not too familiar with nativefier)

**Comment by Blatman** [comment]
FWIW - I just tried Tauri with a static archived web using HTML pages, standard linked JS files and href linked pages (relative paths) and it worked fine. Just used npm create tauri-app and specified new app name, used the dist folder as the server root and did vanilla JS (no framework) then dropped all the web pages with images folder, etc into the dist folder and then did a build. Ended up with a working DMG and app worked well. Built it on 12.5.1 as a x64 binary and ran it on 10.13 OK.

If I was looking at bundling a static site from the web I might use something like SiteSucker to obtain the web pages/JS/CSS/images etc and then just drop them into the dist folder.

**Comment by huangjj27** [comment]
A simple way to do this, is to set `window.location.href` to websites you want to bundle, which is entirely not of Tauri's business

**Comment by hamza72x** [comment]
Check this: https://github.com/hamza72x/web2app
I think this is what you are looking for

@RpxdYTX

**Reply by MakeDeeply** [reply]
@hamza72x web2app looks cool. Would be useful to add a license file and check the appropriate box in Github. Ideally MIT, BSD or Apache. :)

**Reply by hamza72x** [reply]
@MakeDeeply https://github.com/hamza72x/web2app/blob/main/LICENSE
